python/py-parse-type/Portfile

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           python 1.0
+
+github.setup        jenisys parse_type 0.5.2 v
+revision            0
+name                py-parse-type
+categories-append   devel
+
+license             BSD
+maintainers         nomaintainer
+platforms           darwin
+
+description         parse_type extends the parse module\
+                    (opposite of string.format())
+
+long_description    parse_type extends the parse module (opposite of\
+                    string.format()) with the following features:\
+                    build type converters for common use cases\
+                    (enum/mapping, choice)\; build a type converter\
+                    with a cardinality constraint (0..1,0..*,1..*)\;\
+                    from the type converter with cardinality=1.\;\
+                    compose a type converter from other type\
+                    converters\; an extended parser that supports the\
+                    CardinalityField naming schema and creates missing\
+                    type variants (0..1,0..*,1..*) from the primary\
+                    type converter
+
+checksums           rmd160  4b1cedd0e758ee224829848f45704578f55dc0b2 \
+                    sha256  eec72a1a3d48f6a4196c28a614151e3968283726e52604fc1cdf4d9e597985fe \
+                    size    268927
+
+python.versions     37 38
+
+if {${name} ne ${subport}} {
+    depends_build-append \
+                    port:py${python.version}-setuptools
+
+    depends_run-append \
+                    port:py${python.version}-parse \
+                    port:py${python.version}-six
+
+    depends_test-append \
+                    port:py${python.version}-pytest
+
+    post-destroot {
+        set docdir ${prefix}/share/doc/${subport}
+        xinstall -d ${destroot}${docdir}
+        xinstall -m 0644 -W ${worksrcpath} README.rst LICENSE \
+            ${destroot}${docdir}
+    }
+
+    test.run        yes
+    test.env        PYTHONPATH=${worksrcpath}/build/lib
```

Comment:
again, ```sudo port -dv test py37-parse-type``` does not work - please fix that now and in the future do test this before submitting a PR.

Reply:
??? This one works for me:
```

tests/test_parse_util.py ................... [ 93%]
..............                               [100%]

=============== 207 passed in 0.54s ================
```

Reply:
no, it does not: ```error: invalid command 'pytest'```. If this works for you there are two options: (i) you either installed ```pytest``` installed by other means than MacPorts, or (i)) used the ```select``` to set a specific version. The latter can be checked by doing ```port select --list pytest```, if this shows anything other than "none" as being active that would explain it. In any case, both of these options cannot be relied on.

Reply:
I don’t have it set:
```
port select --show pytest
The currently selected version for 'pytest' is 'none'.
which pytest
<nothing>
```

I don’t see the error you see, and `port -dv test py37-parse-type ` runs successfully.